torch/_library/custom_ops.py

```diff
@@ -275,6 +277,78 @@ def register_fake(self, fn: Callable) -> Callable:
         self._abstract_fn = fn
         return fn
 
+    def register_autograd(
+        self, setup_context_fn: Callable, backward_fn: Callable, /
+    ) -> None:
+        r"""Register a backward formula for this custom op.
+
+        In order for an operator to work with autograd, you need to register
+        a backward formula. There are two pieces to this:
+        1. You must tell us what we need to save from the forward pass for
+           the backward pass. This is the "setup_context" function.
+        2. You must tell us how to compute gradients during the backward pass.
+           This is the "backward" function.
+
+        ``setup_context_fn(ctx, inputs, output)`` runs during the forward pass.
+        Please save quantities needed for backward onto the ``ctx`` object via
+        either :func:`ctx.save_for_backward` or assigning them as attributes of
+        ``ctx``.
+
+        ``backward_fn`` runs during the backward pass. It accepts ``(ctx, *grads)``:
+        - ``grads`` is one or more gradients. The number of gradients matches
+          the number of outputs of the operator.
+
+        Both ``setup_context_fn`` and ``backward_fn`` must be traceable. That is,
+        they may not directly access Tensor.data_ptr and they must not depend on
+        or mutate global state.
+
+        Examples:
+            >>> import torch
+            >>> import numpy as np
+            >>> from torch import Tensor
+            >>>
+            >>> @torch.library.custom_op("mylib::numpy_sin", mutated_args=())
+            >>> def numpy_sin(x: Tensor) -> Tensor:
+            >>>     x_np = x.cpu().numpy()
+            >>>     y_np = np.sin(x_np)
+            >>>     return torch.from_numpy(y_np).to(device=x.device)
+            >>>
+            >>> def setup_context(ctx, inputs, output) -> Tensor:
+            >>>     x, = inputs
+            >>      ctx.save_for_backward(x)
+            >>>
+            >>> def backward(ctx, grad):
+            >>>     x, = ctx.saved_tensors
+            >>>     return grad * x.cos()
+            >>>
+            >>> numpy_sin.register_autograd(setup_context, backward)
```

Comment:
Taking API bikeshedding feedback

Reply:
One option that comes to mind since setup_context is optional is to have this decorator be around backward_fn and take in only setup_context?